core/src/databases_store/gcs.rs

```diff
@@ -20,6 +20,42 @@ use super::store::DatabasesStore;
 #[derive(Clone)]
 pub struct GoogleCloudStorageDatabasesStore {}
 
+pub async fn write_rows_to_bucket(
+    schema: &TableSchema,
+    rows: &Vec<Row>,
+    bucket: &str,
+    bucket_csv_path: &str,
+) -> Result<(), anyhow::Error> {
+    let mut field_names = schema
+        .columns()
+        .iter()
+        .map(|c| c.name.clone())
+        .collect::<Vec<_>>();
+
+    // Read all rows and upload to GCS
+    let mut wtr = Writer::from_writer(vec![]);
+
+    // We need to append the row_id in a __dust_id field
+    // It's important to have it LAST in the headers as the table schema do not show it
+    // and when the csv will be used as-is for the sqlite database, the dust_id will be completely ignored (as it should).
+    // See Row.to_csv_record
+    field_names.push("__dust_id".to_string());
+
+    // Write the header.
+    wtr.write_record(field_names.iter().map(String::as_str))?;
```

Comment:
not familiar with this API -- it's not async so is it doing anything CPU expensive ?